src/math/p5.Vector.js

```diff
@@ -584,22 +602,31 @@ p5.Vector.prototype.sub = function sub(x, y, z) {
  * </code>
  * </div>
  */
-p5.Vector.prototype.mult = function mult(n) {
-  if (!(typeof n === 'number' && isFinite(n))) {
-    console.warn(
-      'p5.Vector.prototype.mult:',
-      'n is undefined or not a finite number'
-    );
+p5.Vector.prototype.mult = function mult(x, y, z) {
+  if (x instanceof p5.Vector) {
+    this.x *= x.x || 0;
+    this.y *= x.y || 0;
+    this.z *= x.z || 0;
```

Comment:
Why is the type check removed and why does it default to 0 here if falsy? If the value evaluates to falsy (`NaN` probably), it is easier to debug than if it got defaulted to 0.

Reply:
I was just following the pattern that was already existing in `p5.Vector.add` and `p5.Vector.sub`, neither of which[ do the type checking](https://github.com/processing/p5.js/blob/master/src/math/p5.Vector.js#L310) . I can add the type check back in if you think it's helpful.

Reply:
@limzykenneth I added back in some type checking. it's a bit more 
verbose but I think should cover all the cases (that I could think of)